src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,230 @@
+const Papa = require('papaparse')
+const fs = require('fs')
+
+const ApiHandler = require('../ApiHandler.class')
+
+/**
+ * Class CsvToHttp - convert a csv file into http request such as POST/PUT/PACTH
+ */
+class CsvToHttp extends ApiHandler {
+  /**
+   * Constructor for CsvToHttp
+   * @constructor
+   * @param {Object} applicationParameters - The application parameters
+   * @param {Engine} engine - The Engine
+   * @return {void}
+   */
+  constructor(applicationParameters, engine) {
+    super(applicationParameters, engine)
+
+    const { applicativeHostUrl, requestMethod, authentication, csvDelimiter, mapping, proxy = null } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {}
+    this.request.host = applicativeHostUrl
+    this.request.method = requestMethod
+    this.request.authentication = authentication
+
+    this.mapping = mapping || {}
+    this.mapping.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.httpBody = []
+    this.canHandleFiles = true
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format, file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    await this.parseAndMapp(filePath)
+
+    if (this.httpBody.length !== 0) {
+      await this.sendData()
+    }
```

Comment:
If the body is empty, it comes from the configuration of the user and the file. So I don't think an error is best approach. Maybe a warning but not sure it will be useful.

sendData return a promise, so if the sendRequest method fails it will not return a success.